client/types.go

```diff
@@ -0,0 +1,155 @@
+package client
+
+import (
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+)
+
+// BaseReq defines a structure that can be embedded in other request structures
+// that all share common "base" fields.
+type BaseReq struct {
+	From          string       `json:"from"`
+	Password      string       `json:"password"`
+	Memo          string       `json:"memo"`
+	ChainID       string       `json:"chain_id"`
+	AccountNumber uint64       `json:"account_number"`
+	Sequence      uint64       `json:"sequence"`
+	Fees          sdk.Coins    `json:"fees"`
+	GasPrices     sdk.DecCoins `json:"gas_prices"`
+	Gas           string       `json:"gas"`
+	GasAdjustment string       `json:"gas_adjustment"`
+	GenerateOnly  bool         `json:"generate_only"`
+	Simulate      bool         `json:"simulate"`
+}
+
+// SendReq requests sending an amount of coins
+type SendReq struct {
+	Amount  sdk.Coins `json:"amount"`
+	BaseReq BaseReq   `json:"base_req"`
+}
+
+// MsgDelegationsInput requests a delegation
+type MsgDelegationsInput struct {
```

Comment:
No problem, merge mine first ;) hehe

Reply:
(joking) I did this because it was requested as part of the code review.
I see that you did something similar yesterday. It was definitely necessary.

Reply:
As discussed with Alessio, I will merge changes made in https://github.com/cosmos/cosmos-sdk/pull/3487 into this PR

Reply:
#3487 seems ready to go. I will wait until is merged and will rebase.